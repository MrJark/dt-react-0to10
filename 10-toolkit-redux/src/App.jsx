import { useDispatch, useSelector } from 'react-redux';


import reactLogo from './assets/react.svg';
import viteLogo from '/vite.svg';
import './App.css';
import { increment, decrement, incrementBy } from './store/slices/counter';

function App() {

  const { counter } = useSelector( state => state.counter);
  const dispatch = useDispatch();
  const numByIncrement = 2;
  
  return (
    // tarea: hacer el decrement function y el increment by 2 donde el 2 tiene que ser un argumento que se pueda modificar dependiendo del incremento. Que no sea estático
    // incremento ✅ y el incrementBy ❌ me salía el value como null y no sabía como cambiarlo
    // lo había hecho bien solo que al copiar del ejemplo, no había cambiado el value por el counter y claro, no existía
    <>
      <div>
        <a href="https://vitejs.dev" target="_blank">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://react.dev" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h2>count is {counter}</h2>
      <div className="card">
        <button type='button' style={{ margin: 4}} onClick={() => {dispatch( increment() )}}>
          Increment
        </button>
        <button type='button' style={{ margin: 4}} onClick={() => {dispatch( decrement() )}}>
          Decrement
        </button>
        <button type='button' style={{ margin: 4}} onClick={() => {dispatch( incrementBy(numByIncrement) )}}>
          Increment by {numByIncrement}
        </button>
      </div>
    </>
  )
}

export default App
