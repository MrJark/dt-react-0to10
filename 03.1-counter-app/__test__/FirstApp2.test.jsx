import {render, screen} from '@testing-library/react';
// el render es el método y el scrren es la pantalla, el window donde está el contenido
import App from '../src/FirstApp';

// Simplificación del código de FistApp

describe('pruebas en <App /> de FirstApp2', () => {

    const title = 'Hola, soy Vegeta';
    const subTitle = 'Soy un subitulo';

    test('debe hacer match con el snapshot', () => {

        const { container } = render( <App title={title} />);
        expect( container ).toMatchSnapshot();
    });

    test('debe mostrar el mensaje "Hola soy Vegeta" ', () => {

        render( <App title={title} />);
        expect( screen.getByText(title)).toBeTruthy();
        // screen.debug(); // para ver el objeto
    });

    test('debe mostrar el titulo como h1', () => {

        render( <App title={title} />);
        expect( screen.getByRole('heading', {level: 1}).innerHTML).toContain( title ); // en el screen, que tenga el rol de header con nivel 1 (h1) y que esté en le html y que eso, ese elemento, contenga el tiitulo aun pudiendo tener espacios

    });

    test('debe mostrar el subtitulo enviado por props', () => {

        render(
            <App 
                title={title} 
                subTitle={subTitle}
            />
        );
        expect( screen.getAllByText(subTitle).length).toBe(1);

    })
})