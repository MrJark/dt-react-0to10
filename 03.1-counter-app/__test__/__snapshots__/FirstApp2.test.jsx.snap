// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <App /> de FirstApp2 debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Vegeta
  </h1>
  <h2>
    Soy un subtitulo
  </h2>
  <h2>
    Chema Ferrandez
  </h2>
</div>
`;
